### ISSUE TITLE
AverageUnpooling layer for PyTorch (Proposal)

### ISSUE THREAD

**Issue by revanurambareesh** [pytorch/pytorch#19805]
## 🚀 Feature
PyTorch supports MaxPooling, AveragePooling layers and also supports MaxUnpooling layer. However AverageUnpooling layer is currently not supported in PyTorch.

## Motivation
As a researcher I find that having AverageUnpooling layer as part of PyTorch will be very useful for following reasons:
1. Common Use-case: CNN autoencoder with average pooling. Other use-cases include image reconstruction.
2. The average unpooling layer is also used (although not as extensively as MaxUnpooling) in research and I strongly feel that this layer should exist in PyTorch.
3. This will also help address average unpooling queries, such as one posted here in [discuss.pytorch.org](https://discuss.pytorch.org/t/the-most-recommended-way-to-perform-average-unpooling-in-pytorch/10705)

## Pitch

**Implementing AverageUnpooling layer** similar to https://github.com/HyeonwooNoh/caffe/blob/master/src/caffe/layers/unpooling_layer.cpp

EDIT: PyTorch's interpolation suits my requirements. Hence closed the issue.

**Comment by fmassa** [comment]
Isn't `AverageUnpooling` very similar to `F.interpolate`?

**Comment by zou3519** [comment]
What exactly are the semantics of `AverageUnpooling`?

**Comment by revanurambareesh** [comment]
Yeah, It is very similar to `F.interpolate`, but with padding. I was thinking that having it as a layer with the name `AvgUnpool` would be beneficial. However, since it is semantically similar to interpolation, I am closing the issue. Pytorch's interpolation suffices my requirements. Thanks!
